### PR TITLE
Add tests for SQLXML.

### PR DESCRIPTION
**Pull request by wiarlawd**
Oracle and SQL Server support an XML data type.

### CODE REVIEW

File: test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java [hunk 1]
```diff
@@ -2450,6 +2451,63 @@ public void testMetadataColumns_clobNull() throws Exception {
   }
 
   @Test
+  public void testMetadataColumns_sqlxml() throws Exception {
+    assumeTrue("SQLXML type not supported", is(ORACLE) || is(SQLSERVER));
+    String content = "<motd>hello world</motd>\n";
+    executeUpdate("create table data(id int, content xml)");
+    String sql = "insert into data(id, content) values (1, ?)";
+    try (Connection conn = getConnection();
```
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java:2459]: Use JdbcFixture.prepareStatement and you can lose the try-with-resources.

File: test/com/google/enterprise/adaptor/database/ResponseGeneratorTest.java [hunk 2]
```diff
@@ -464,6 +467,49 @@ public void testContentColumn_nullBlob() throws Exception {
   }
 
   @Test
+  public void testContentColumn_sqlxml() throws Exception {
+    assumeTrue("SQLXML type not supported", is(ORACLE) || is(SQLSERVER));
+    String content = "<motd>hello world</motd>\n";
+    executeUpdate("create table data(id int, content xml)");
+    String sql = "insert into data(id, content) values (1, ?)";
+    try (Connection conn = getConnection();
```
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/database/ResponseGeneratorTest.java:475]: Ditto with JdbcFixture.prepareStatement.

File: test/com/google/enterprise/adaptor/database/TupleReaderTest.java [hunk 3]
```diff
@@ -769,6 +772,87 @@ public void testBlob_null() throws Exception {
   }
 
   @Test
+  public void testSqlxml() throws Exception {
+    assumeTrue("SQLXML type not supported", is(ORACLE) || is(SQLSERVER));
+    String xmlData = "<motd>hello world</motd>\n";
+    executeUpdate("create table data(COLNAME xml)");
+    String sql = "insert into data(colname) values (?)";
+    try (Connection conn = getConnection();
```
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/database/TupleReaderTest.java:780]: ditto

File: test/com/google/enterprise/adaptor/database/TupleReaderTest.java [hunk 4]
```diff
@@ -769,6 +772,87 @@ public void testBlob_null() throws Exception {
...
+    assertEquals(golden, result);
+  }
+
+  @Test
+  public void testSqlxml_empty() throws Exception {
+    // Oracle does not support empty XMLTYPE values.
+    assumeTrue("SQLXML type not supported", is(SQLSERVER));
+    executeUpdate("create table data(COLNAME xml)");
+    String sql = "insert into data(colname) values (?)";
+    try (Connection conn = getConnection();
```
Comment (BrettMichaelJohnson) [inline test/com/google/enterprise/adaptor/database/TupleReaderTest.java:812]: and again

File: test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java (hunk 1 above)
Comment (wiarlawd) [reply test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java:2459]: I need the Connection to create the SQLXML object.

Comment (BrettMichaelJohnson) [reply test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java:2459]: missed that.

**Review by BrettMichaelJohnson — APPROVED** [review]

**Pull request merged**